### PR TITLE
Updates frontend to support multiple pnl reports and pagination

### LINKED ISSUE #1214: Store PnL reports in the DB
## Problem Definition

In [this](https://github.com/rotki/rotki/issues/1212#issue-665581857) issue @davbre mentioned that it may be beneficial for users to store tax reports and explore them again later without having to regenerate them.

## Task

1. In theory if all caching works fine re-generating a tax report for a range that has already been explored should be fast, so test that and make sure that it's fast and if not figure out why and what can be done to speed it up.
2. Think about it and how such a feature would look like

### LINKED ISSUE #3641: Frontend timeouts while trying to parse very big json response
## Problem Definition

While trying to create the PnL report for a big number of events the frontend hangs and no output is produced

## How to reproduce

In the backend change https://github.com/rotki/rotki/blob/48589c7d79e8a464c9f44084bd6ce4a1a3d9730c/rotkehlchen/accounting/accountant.py#L466 to something like `'all_events': self.csvexporter.all_events*10000,` and request a full year of events (or a period with enough events).

## Observable behavior

The frontend successfully retrieves the task response from the backend but it timeouts trying to read the response with errors in the console about the request failing but not providing any further information

### PR DESCRIPTION
**Pull request by kelsos**
Closes #1214
Closes #3641

## Checklist

- [x] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by kelsos** [comment]
`npm ci` is needed before here since it introduces `pinia` as a dependency

**Review by LefterisJP — COMMENTED** [review]
> `npm ci` is needed before `` npm run dev`` here since it introduces ``pinia` as a dependency

haha ... yeah I noticed the minute I tried to run it.

Also seems to need a rebase to solve conflicts?

Is 100 files changes normal? I guess due to pinia?

**Comment by kelsos** [comment]
> > `npm ci` is needed before ` npm run dev` here since it introduces ``pinia` as a dependency
> 
> haha ... yeah I noticed the minute I tried to run it.
> 
> Also seems to need a rebase to solve conflicts?
> 
> Is 100 files changes normal? I guess due to pinia?

If I only touched the PNL report part it would not be normal, 
That is only due to rewriting 3 vuex modules to pinia.

Don't worry about the conflicts, I can fix them later. At the moment we need to see if there is something that needs to change.

**Review by LefterisJP — COMMENTED** [review]
Apart from the things we talked about in discord this needs two changelog entries:

1. feature: https://github.com/rotki/rotki/issues/1214
2. bug: https://github.com/rotki/rotki/issues/3641

And also updated docs with screenshots to showcase the new functionality.

**Review by LefterisJP — COMMENTED** [review]
Changes for CSV export look good. As mentioned still need docs and changelog

File: docs/changelog.rst [hunk 1]
```diff
@@ -10,6 +11,7 @@ Changelog
 * :feature:`3725` Users can now see if any of their addresses have SDL available to claim from the SDL airdrop.
 * :feature:`3549` Users will now be able to select whether to include or not NFT total value in total net worth and graphs.
 * :feature:`3712` Users will now be able to choose a custom date format while importing CSV files.
+* :bug:`3641` Users should now be able to access big profit and loss report without errors.
```
Comment (LefterisJP) [inline docs/changelog.rst:14]:
```suggestion
* :bug:`3641` Users should now be able to view profit and loss reports that have many thousands of events without the UI freezing.
```

**Review by LefterisJP — APPROVED** [review]
LGTM. Small wording adjustment on changelog. Can be done in a later PR.

**Pull request merged**